### ISSUE TITLE
Invalid YAML workflow files silently pass CI checks

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#107]
## Description

When a GitHub Actions workflow file contains invalid YAML syntax, the `process.ci` check silently passes instead of reporting the parsing error.

## Reproduction

1. Create an invalid workflow file:
```yaml
# .github/workflows/ci.yml
name: Invalid YAML
jobs:
  test
    steps:
```
(Note: missing colon after `test`)

2. Configure check.toml:
```toml
[process.ci]
enabled = true
require_workflows = ["ci.yml"]
```

3. Run `cm process check`

## Expected Behavior

The check should fail and report a YAML parsing error.

## Actual Behavior

The check passes with "✓ CI: passed". The invalid YAML file is silently skipped.

## Impact

Users won't be notified that their workflow files have syntax errors, leading to broken CI pipelines.

## Suggested Fix

In `src/process/tools/ci.ts`, the `parseWorkflow()` method should report parsing errors instead of returning `null` silently. When a required workflow file exists but can't be parsed, this should be treated as a violation.

## Environment

- check-my-toolkit version: 1.3.0
